include/functions_csr.php

```diff
@@ -476,7 +476,7 @@ function sign_csr($passPhrase,$my_csrfile,$my_days,$my_device_type) {
 $config=$_SESSION['config'];
 $name = base64_encode(substr($my_csrfile, 0,strrpos($my_csrfile,'.')));
 $ext = substr($my_csrfile, strrpos($my_csrfile,'.'));
-$my_base64_csrfile=$name.$ext;
+$my_base64_csrfile = $my_csrfile //$name.$ext;

```

Comment:
@JonTheNiceGuy 
OK, all filenames are stored in base64 format, except the extension. The extension is stored as either .pem or .p12, in the relevant folders.
2 reasons for base64 encoding the filenames
1. for backward compatibility with tinyca, as this project was originally developped to use the certstore from tinyca
2. to allow for the cert DN to be imploded and then converted into base64, as the filename, so, files like the private key, which cannot be interrogated for the cert DN, can be stored with a filename, which is unique.

As you see from the original code snippet, i extract the filename, then convert it to base64, then reattached the extension. the extension is reattached with the . and ext, since the strpos, gives the location of the . 

By truncating the base64 filenames down to 32 characters, you run the risk of never finding the private keys, or worse, a hashing overlap, when encoding 2 common names, to the same base64 name, as you have truncated the extension off.

As such, at this time, I cannot include the Windows branch changes into the master, unless, we
1. come up with a method of storing the private key name, uniquely, within a 32 character hash.
2. modify the code, so that when display the cert dn, we interrogate the certificates and csr objects to determine the DN details, instead of asserting them from the filename.